### PR TITLE
Config pages: Redesign

### LINKED ISSUE #4776: Configuration reloading - add more reload services, redesign card
## The request

Configuration reloading card under Server Control panel is currently missing a bunch of things since a lot more reload services have been added lately (input_*, zone, etc). Although it doesn't necessarily require a redesigned card to add these, the current style would result in a very tall height.

Here is an idea I had:

reload-config

Chips with multi-select would allow adding all of the various services while still keeping the card a reasonable size. Even with everything added, it would be shorter than it currently is (probably). An added benefit is that you can reload multiple things at once. So if you make a brand new automation that has an input boolean you could do them both at the same time.

## The alternatives

Alternative would be to keep the existing style but just add everything to it, which would result in a super tall card. Below is a mockup of that, and it's missing stuff like Reload Zones which was just added recently (so it'd be even taller over time). 

tall

### LINKED ISSUE #5511: Configuration Menu Reorganization
## The request
In the Configuration menu, I think it would be better to group the “Customizations” submenu with “Integrations, Devices, Entities and Areas”. It’s just a cosmetic or UX detail, but in my opinion, it would make more sense.

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

image
image
image
image
image

### CODE REVIEW

**Comment by zsarnett** [comment]
I need to pull the changes from Ludeeus's Progress button.

**Comment by zsarnett** [comment]
These are some big changes. It affects a lot of pages. Only pages I believe I was unable to check was the Zwave, OZW pages.

I still don't necessarily love the YAML reloading but it does use the space better and there are going to be a lot more in 115

**Comment by bramkragten** [comment]
For the YAML reloading, let's just use normal buttons with columns.

image

**Comment by bramkragten** [comment]
If we place 2 cards side by side, let's make sure they are the same height:

image

**Comment by bramkragten** [comment]
Make sure the input fields are big enough on all view ports, side by side is too narrow in the screenshot below.

image

**Comment by zsarnett** [comment]
For the buttons, if we use regular buttons we don't get the success feedback or error feedback. I am not sure if any these are necessary for these actions though

**Comment by bramkragten** [comment]
> For the buttons, if we use regular buttons we don't get the success feedback or error feedback. I am not sure if any these are necessary for these actions though

I meant for the styling mainly, you could add the right properties/styles to the call API button so it can be outlined and have a width.

But normal button could also work.

**Comment by noxhirsch** [comment]
> I still don't necessarily love the YAML reloading but it does use the space better and there are going to be a lot more in 115

I think on the YAML reload section, the redundand "RELOAD" on every button should be removed - that way it looks cleaner and make it easier to find the button you need. I made a mockup with Photoshop:
image
The text above could then say something like 
"[...] Hitting one of the buttons below will unload the current YAML configuration of that domain and load the new one." 
or shorter 
"[...] Hitting one of the buttons below will reload the YAML configuration of that domain."

**Comment by zsarnett** [comment]
image
image
image
image

**Comment by noxhirsch** [comment]
Really nice! 👍
Maybe to make the group button shorter and more like the location button: "groups, group entities & notify" (I think it's clear without "services")

File: src/translations/en.json [hunk 1]
```diff
@@ -826,38 +826,38 @@
...
-              "history_stats": "Reload history stats entities",
-              "trend": "Reload trend entities",
-              "ping": "Reload ping binary sensor entities",
-              "filesize": "Reload file size entities",
-              "telegram": "Reload telegram notify services",
-              "smtp": "Reload smtp notify services",
-              "mqtt": "Reload mqtt entities",
-              "rpi_gpio": "Reload Raspberry Pi GPIO entities"
+              "introduction": "Some parts of Home Assistant can reload without requiring a restart. Clicking a button below will unload their current YAML configuration and load the new one.",
+              "reload": "{domain}",
```
Comment (bramkragten) [inline src/translations/en.json:830]: This doesnt make sense anymore now 😄

**Comment by zsarnett** [comment]
image

**Comment by zsarnett** [comment]
TODO: 

- [ ] Localize Everything
- [ ] Add Person secondary info (Was last at, or something else if not device tracker)
- [ ] Link to from individual items in Helper, Tags, and Lovelace Dashboards (link to the popup editor?)
- [ ] Link to from cloud items
- [ ] Link to from integration items

Questions: 

- Does this seem like a good implementation? I am querying a lot of data for this page.

**Comment by Misiu** [comment]
@zsarnett I think the title `Hey Zack, Welcome Home!` is unneeded.
How will this look if you have more integrations, automation, scripts, and other items?
I have about 30 automations. Maybe show last edited (two or three) and add a count in the title, something like this:
image

**Comment by zsarnett** [comment]
The title will be changed meant to add that to the list. And in the code you should see the integrations are being sliced

File: src/panels/config/dashboard/ha-config-dashboard.ts [hunk 2]
```diff
@@ -134,45 +124,888 @@ class HaConfigDashboard extends LitElement {
...
+
+  private _getHelpers = memoizeOne(
+    (states: HassEntities): AutomationEntity[] => {
+      return Object.values(states)
+        .filter((entity) => HELPER_DOMAINS.includes(computeStateDomain(entity)))
+        .slice(0, 2) as AutomationEntity[];
+    }
+  );
+
+  private async _fetchTags() {
```
Comment (Misiu) [inline src/panels/config/dashboard/ha-config-dashboard.ts:739]:
```suggestion
  private async _getTags() {
```
maybe rename to make this consistent with _getScenes, _getScripts, and others

**Comment by Misiu** [comment]
> The title will be changed meant to add that to the list. And in the code you should see the integrations are being sliced

Right, I see slice(0,2) :)
But what about a counter light to the section name?

**Comment by zsarnett** [comment]
Yea. I plan to bring something like that to those sections. I do like the idea of it being to the right. Not sure if we should put a (+) there or the number.

image


This is the reference

**Comment by Misiu** [comment]
> Not sure if we should put a (+) there or the number.

Plus sign suggests an add action (it should navigate you to new automation/script page). I personally would prefer a number showing the number of automations/scripts etc.

**Comment by zsarnett** [comment]
Right. The plus sign would be the add action :)

**Comment by zsarnett** [comment]
Yea yea. I know bot. I'll do something with this eventually

**Comment by bramkragten** [comment]
If you ever get back to this, we should really break it up into smaller PR's

**Pull request closed without merging**